lib/src/parsers/utilities/parser-utility.ts

```diff
@@ -281,11 +306,31 @@ export function extractDecoratorFactoryConfiguration(
     }
   } else {
     throw new Error(
-      `expected 1 argument, got ${decorator.getArguments().length}`
+      `expected at least 1 argument, got ${decorator.getArguments().length}`
     );
   }
 }
 
+/**
+ * Extract the Options object (second argument) from a Spot decorator factory.
+ *
+ * @param decorator a decorator factory
+ */
+export function extractDecoratorFactoryOptions(
+  decorator: Decorator
+): ObjectLiteralExpression | undefined {
+  if (decorator.getArguments().length >= 2) {
+    const argument = decorator.getArguments()[1];
+    if (TypeGuards.isObjectLiteralExpression(argument)) {
+      return argument;
+    } else {
+      throw new Error(`expected object literal`);
+    }
+  } else {
+    return undefined;
```

Comment:
How about returning `null` instead? I generally think of `undefined` as not returning anything, whereas here you're returning the absence of a value.